Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs

```diff
@@ -1,43 +1,64 @@
 ﻿using Promact.Trappist.DomainModel.DbContext;
 using System.Linq;
 using Promact.Trappist.DomainModel.Models.Test;
-using System.Collections.Generic;
+using System;
 using System.Threading.Tasks;
+using System.Collections.Generic;
 using Microsoft.EntityFrameworkCore;
+using Promact.Trappist.Utility.Constants;
 
 
 namespace Promact.Trappist.Repository.Tests
 {
     public class TestsRepository : ITestsRepository
     {
+        private static Random random = new Random();
         private readonly TrappistDbContext _dbContext;
-        public TestsRepository(TrappistDbContext dbContext)
+        private readonly IStringConstants _stringConstants;
+        public TestsRepository(TrappistDbContext dbContext, IStringConstants stringConstants)
         {
             _dbContext = dbContext;
+            _stringConstants = stringConstants;
         }
         /// <summary>
         /// this method is used to create a new test
         /// </summary>
         /// <param name="test">object of Test</param>
-        public void CreateTest(Test test)
+        public async Task CreateTestAsync(Test test)
         {
             _dbContext.Test.Add(test);
-            _dbContext.SaveChangesAsync();
+             await _dbContext.SaveChangesAsync();            
+        }
+        /// <summary>
+        /// Fetch all the tests from Test Model,Convert it into List
+        /// </summary>
+        /// <returns>List of Tests</returns>   
+        public List<Test> GetAllTests()
+        {
+            var tests = _dbContext.Test.ToList();
+            return tests;
         }
+
+        /// <summary>
+        ///  this method is used to generate a random string for every test at the time of test creation
+        /// <param name="test">object of Test</param>
+        /// <param name="length">length of the random string</param>
+        /// </summary>
+        public void RandomLinkString(Test test, int length)
+        {
+            string charactersForRandomString = _stringConstants.CharactersForLink;
+            test.Link = new string(Enumerable.Repeat(charactersForRandomString, length).Select(s => s[random.Next(s.Length)]).ToArray());
+        }
+
         /// <summary>
         /// this method is used to check whether test name is unique or not
         /// </summary>
         /// <param name="test">object of Test</param>
         /// <returns>boolean</returns>
-        public bool UniqueTestName(Test test)
+        public async Task<bool> IsTestNameUniqueAsync(string testName)
         {
-            var testObj = (from s in _dbContext.Test
-                           where s.TestName == test.TestName
-                           select s).FirstOrDefault();
-            if (testObj != null)
-                return false;
-            else
-                return true;
+            bool testNameCheck = await _dbContext.Test.AnyAsync(x => x.TestName.ToLower() == testName.ToLower());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define the locale to be used in this string operation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1449)